Tests/test_image.py

```diff
@@ -117,6 +117,13 @@ def test_open_formats(self) -> None:
                 assert im.mode == "RGB"
                 assert im.size == (128, 128)
 
+    def test_open_verbose_failure(self) -> None:
+        im = io.BytesIO(b"")
+        with pytest.warns(UserWarning):
+            with pytest.raises(UnidentifiedImageError):
+                with Image.open(im, warn_possible_formats=True):
```

Comment:
I think these can all be in the same `with` statement, separated by commas. Though that might make the line too long.

Reply:
If I do that, and then run black, I get
```python
with pytest.warns(UserWarning), pytest.raises(
    UnidentifiedImageError
), Image.open(im, warn_possible_formats=True):
```
This does not look clearer to me.